src/wallet/wallet2.cpp

```diff
@@ -2655,10 +2656,12 @@ bool wallet2::store_keys(const std::string& keys_file_name, const epee::wipeable
   crypto::chacha20(account_data.data(), account_data.size(), key, keys_file_data.iv, &cipher[0]);
   keys_file_data.account_data = cipher;
 
+  m_keys_file_locker.reset();
   std::string buf;
   r = ::serialization::dump_binary(keys_file_data, buf);
   r = r && epee::file_io_utils::save_string_to_file(keys_file_name, buf); //and never touch wallet_keys_file again, only read
   CHECK_AND_ASSERT_MES(r, false, "failed to generate wallet keys file " << keys_file_name);
+  m_keys_file_locker.reset(new tools::file_locker(m_keys_file.c_str()));
```

Comment:
This uses c_str, but the ctor uses std::string, so this creates a new temporary string for no reason.
Having the ctor use const char \* seems the way to support both without temp object creation.

Reply:
ok